db/migrate/20160214115800_inline_ems_id.rb

```diff
@@ -0,0 +1,51 @@
+class InlineEmsId < ActiveRecord::Migration
+  class ContainerDefinition < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class ContainerGroup < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class Container < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :container_definition, :class_name => "InlineEmsId::ContainerDefinition"
+    has_one    :container_group, :through => :container_definition, :class_name => "InlineEmsId::ContainerGroup"
+  end
+
+  class ContainerDefinition < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :container_group, :class_name => "InlineEmsId::ContainerGroup"
+  end
+
+  def self.up
+    add_columns
+    update_columns
+  end
+
+  def add_columns
```

Comment:
I am amazed this works since the up and down are class methods, but these are instance methods.  Can you verify it works?